File: SchoolManagementApi/DTOs/Teachers/TeacherResponseDto.cs
namespace SchoolManagementApi.DTOs.Teachers;

public class TeacherResponseDto
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Surname { get; set; } = null!;
    public string Email { get; set; } = null!;

    public string FullName => $"{Name} {Surname}";
}

File: SchoolManagementApiNode/src/teachers/dto/teacher-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class TeacherResponseDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  name: string;

  @ApiProperty()
  surname: string;

  @ApiProperty()
  email: string;

  @ApiProperty()
  fullName: string;
}
